src/typing/macroContext.ml

```diff
@@ -769,61 +765,55 @@ let load_macro_module mctx com cpath display p =
+	let timer_level = Timer.level_from_define com.defines Define.MacroTimes in
+	try
+		mctx.com.cached_macros#find (cpath,fname)
+	with Not_found ->
+		let f () =
+			let mpath, sub = (match List.rev (fst cpath) with
+				| name :: pack when name.[0] >= 'A' && name.[0] <= 'Z' -> (List.rev pack,name), Some (snd cpath)
+				| _ -> cpath, None
+			) in
+			let mloaded,restore = load_macro_module mctx com mpath display p in
+			let cl, meth =
+				try
+					if sub <> None || mloaded.m_path <> cpath then raise Not_found;
+					match mloaded.m_statics with
+					| None -> raise Not_found
+					| Some c ->
+						Finalization.finalize mctx;
+						c, PMap.find fname c.cl_statics
+				with Not_found ->
+					let name = Option.default (snd mpath) sub in
+					let path = fst mpath, name in
+					let mt = try List.find (fun t2 -> (t_infos t2).mt_path = path) mloaded.m_types with Not_found -> raise_typing_error_ext (make_error (Type_not_found (mloaded.m_path,name,Not_defined)) p) in
+					match mt with
+					| TClassDecl c ->
+						Finalization.finalize mctx;
+						c, (try PMap.find fname c.cl_statics with Not_found -> raise_typing_error ("Method " ^ fname ^ " not found on class " ^ s_type_path cpath) p)
+					| _ -> raise_typing_error "Macro should be called on a class" p
+			in
+			let meth = (match follow meth.cf_type with TFun (args,ret) -> (args,ret,cl,meth),mloaded | _ -> raise_typing_error "Macro call should be a method" p) in
+			restore();
+			if not com.is_macro_context then flush_macro_context mint mctx;
+			mctx.com.cached_macros#add (cpath,fname) meth;
+			meth
 		in
-		let meth = (match follow meth.cf_type with TFun (args,ret) -> (args,ret,cl,meth),mloaded | _ -> raise_typing_error "Macro call should be a method" p) in
-		restore();
-		if not com.is_macro_context then flush_macro_context mint mctx;
-		mctx.com.cached_macros#add (cpath,f) meth;
-		(* mctx.m <- {
-			curmod = null_module;
-			import_resolution = new resolution_list ["import";s_type_path cpath];
-			own_resolution = None;
-			enum_with_type = None;
-			module_using = [];
-			import_statements = [];
-			is_display_file = false;
-		}; *)
-		t();
-		meth
+		macro_timer com.timer_ctx timer_level "typing" (Some (s_type_path cpath ^ "." ^ fname)) f ()
 
 let load_macro' ctx display cpath f p =
 	(* TODO: The only reason this nonsense is here is because this is the signature
 	   that typer.di_load_macro wants, and the only reason THAT exists is the stupid
 	   voodoo stuff in displayToplevel.ml *)
 	fst (load_macro'' ctx.com (get_macro_context ctx) display cpath f p)
 
-let do_call_macro com api cpath f args p =
-	let t = macro_timer com ["execution";s_type_path cpath ^ "." ^ f] in
+let do_call_macro com api cpath name args p =
 	incr stats.s_macros_called;
-	let r = Interp.call_path (Interp.get_ctx()) ((fst cpath) @ [snd cpath]) f args api in
-	t();
-	r
+	let timer_level = Timer.level_from_define com.defines Define.MacroTimes in
```

Comment:
I'm not sure how that `timer_level` is supposed to work here. It will either be `0` or `1`, but `-D times.macro` will need it to be `2` to show up in `macro.execution` :thinking: